mesonbuild/interpreter/interpreter.py

```diff
@@ -881,6 +882,10 @@ def do_subproject(self, subp_name: str, kwargs: kwtypes.DoSubproject, force_meth
             default_options = [default_options]
         if isinstance(default_options, list):
             default_options = dict((x.split('=', 1) for x in default_options))
+        if extra_default_options:
+            for k, v in extra_default_options.items():
+                if k not in default_options:
+                    default_options[k] = v
```

Comment:
Not sure if it is better, but you could also do: `default_options.setdefault(k, v)`